app/jobs/data_exports/create_job.rb

```diff
@@ -99,38 +122,75 @@ def create_attachment_manifest_file(attachment)
       attachment_manifest_file
     end
 
-    def write_attachments(sample, project, zip)
+    def write_sample_attachments(sample, project, zip)
       sample.attachments.each do |attachment|
-        next if attachment.metadata.key?('associated_attachment_id') && attachment.metadata['direction'] == 'reverse'
+        directory = "#{project.puid}/#{sample.puid}/#{attachment.puid}/#{attachment.file.filename}"
+        write_attachment(directory, zip, attachment)
+      end
+    end
+
+    # Analysis export specific functions---------------------------------------------------------------------
+    def create_analysis_zip(data_export)
+      new_zip_file = Tempfile.new(binmode: true)
+      ZipKit::Streamer.open(new_zip_file) do |zip|
+        workflow_execution = WorkflowExecution.find(data_export.export_parameters['ids'][0])
```

Comment:
**nitpick (blocking):** Can you add in `.includes(outputs: { file_attachment: :blob }, samples_workflow_executions: [ :sample, outputs: { file_attachment: :blob } ])` before the `.find` which should cause all the data you require to be queried at once vs on demand when it is needed. This should speed up processing.

Reply:
Thanks for these, added in 735c6182225cfb1787142006bed5fcf352e61fb3